### ISSUE TITLE
Can't build a project with included Analytics-CSharp library and enabled linker

### ISSUE THREAD

**Issue by igoris** [segmentio/Analytics-CSharp#41]
**Steps to reproduce**:

1. Create a new Xamarin Android project
2. Add Segment.Analytics.CSharp library via NuGet
3. Go to the project settings, set `Linking` to `Sdk Assemblies Only`, save settings
4. Build project

**Actual result**:
Build fails with message
```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Mono.Linker.Steps.XmlResolutionException: Failed to process XML description: resource Segment.Analytics.CSharp.xml in Segment.Analytics.CSharp, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null ---> Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'UnityEngine, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' ---> Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'UnityEngine, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
   at Mono.Cecil.BaseAssemblyResolver.Resolve(AssemblyNameReference name, ReaderParameters parameters) in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/external/cecil/Mono.Cecil/BaseAssemblyResolver.cs:line 172
   at Mono.Linker.AssemblyResolver.Resolve(AssemblyNameReference name, ReaderParameters parameters)
   at Mono.Linker.LinkContext.Resolve(IMetadataScope scope)
   --- End of inner exception stack trace ---
   at Mono.Linker.LinkContext.Resolve(IMetadataScope scope)
   at Mono.Linker.Steps.ResolveFromXmlStep.ProcessAssemblies(LinkContext context, XPathNodeIterator iterator)
   at Mono.Linker.Steps.ResolveFromXmlStep.Process()
   --- End of inner exception stack trace ---
   at Mono.Linker.Steps.ResolveFromXmlStep.Process()
   at Mono.Linker.Steps.BaseStep.Process(LinkContext context)
   at Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
   at Mono.Linker.Pipeline.Process(LinkContext context)
   at MonoDroid.Tuner.Linker.Process(LinkerOptions options, ILogger logger, LinkContext& context)
   at Xamarin.Android.Tasks.LinkAssemblies.Execute(DirectoryAssemblyResolver res)
   at Xamarin.Android.Tasks.LinkAssemblies.RunTask()
   at Microsoft.Android.Build.Tasks.AndroidTask.Execute() in /Users/runner/work/1/s/xamarin-android/external/xamarin-android-tools/src/Microsoft.Android.Build.BaseTasks/AndroidTask.cs:line 17	SegmentErrorReproduce	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Xamarin\Android\Xamarin.Android.Legacy.targets	648	
```

**Expected result**:
Build succeeds

**Sample project that shows issue**: https://github.com/igoris/segment-linker-error

**Environment**:

- Windows 11
- Visual Studio Version 17.3.2

**Comment by wenxi-zeng** [comment]
@igoris thanks for reporting this. will get it fixed asap.

**Comment by wenxi-zeng** [comment]
hey @igoris, just fyi, the issue has been fixed in #42. it is scheduled to be released in our beta version next week along with couple new features. in the meanwhile, please disable linker as a workaround. I'll update to you once the beta version is out. thanks for your patient 😄

**Comment by didiergarcia** [comment]
Fix is shipped in the latest release 2.1.0
